src/analysis/processing/qgsalgorithmzonalminmaxpoint.cpp

```diff
@@ -0,0 +1,191 @@
+{
+  return new QgsZonalMinimumMaximumPointAlgorithm();
+}
+
+void QgsZonalMinimumMaximumPointAlgorithm::initParameters( const QVariantMap &configuration )
+{
+  Q_UNUSED( configuration )
+  addParameter( new QgsProcessingParameterRasterLayer( QStringLiteral( "INPUT_RASTER" ), QObject::tr( "Raster layer" ) ) );
+  addParameter( new QgsProcessingParameterBand( QStringLiteral( "RASTER_BAND" ),
+                QObject::tr( "Raster band" ), 1, QStringLiteral( "INPUT_RASTER" ) ) );
+}
+
+QString QgsZonalMinimumMaximumPointAlgorithm::outputName() const
+{
+  return QObject::tr( "Zonal Extrema" );
+}
+
+QgsFields QgsZonalMinimumMaximumPointAlgorithm::outputFields( const QgsFields &inputFields ) const
+{
+  Q_UNUSED( inputFields )
+  return mOutputFields;
+}
+
+Qgis::ProcessingSourceType QgsZonalMinimumMaximumPointAlgorithm::outputLayerType() const
+{
+  return Qgis::ProcessingSourceType::VectorPoint;
+}
+
+Qgis::WkbType QgsZonalMinimumMaximumPointAlgorithm::outputWkbType( Qgis::WkbType ) const
+{
+  return Qgis::WkbType::Point;
+}
+
+QgsFeatureSink::SinkFlags QgsZonalMinimumMaximumPointAlgorithm::sinkFlags() const
+{
+  return QgsFeatureSink::SinkFlag::RegeneratePrimaryKey;
+}
+
+Qgis::ProcessingAlgorithmDocumentationFlags QgsZonalMinimumMaximumPointAlgorithm::documentationFlags() const
+{
+  return Qgis::ProcessingAlgorithmDocumentationFlag::RegeneratesPrimaryKey;
+}
+
+QgsCoordinateReferenceSystem QgsZonalMinimumMaximumPointAlgorithm::outputCrs( const QgsCoordinateReferenceSystem & ) const
+{
+  return mCrs;
+}
+
+bool QgsZonalMinimumMaximumPointAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback * )
+{
+  QgsRasterLayer *rasterLayer = parameterAsRasterLayer( parameters, QStringLiteral( "INPUT_RASTER" ), context );
+  if ( !rasterLayer )
+    throw QgsProcessingException( invalidRasterError( parameters, QStringLiteral( "INPUT_RASTER" ) ) );
+
+  mBand = parameterAsInt( parameters, QStringLiteral( "RASTER_BAND" ), context );
+  if ( mBand < 1 || mBand > rasterLayer->bandCount() )
+    throw QgsProcessingException( QObject::tr( "Invalid band number for BAND (%1): Valid values for input raster are 1 to %2" ).arg( mBand )
+                                  .arg( rasterLayer->bandCount() ) );
+
+  if ( !rasterLayer->dataProvider() )
+    throw QgsProcessingException( QObject::tr( "Invalid raster layer. Layer %1 is invalid." ).arg( rasterLayer->id() ) );
+
+  mRaster.reset( rasterLayer->dataProvider()->clone() );
+  mCrs = rasterLayer->crs();
+  mPixelSizeX = rasterLayer->rasterUnitsPerPixelX();
+  mPixelSizeY = rasterLayer->rasterUnitsPerPixelY();
+  std::unique_ptr<QgsFeatureSource> source( parameterAsSource( parameters, inputParameterName(), context ) );
+
+  QgsFields newFields;
+  newFields.append( QgsField( QStringLiteral( "value" ), QMetaType::Type::Double, QString(), 20, 8 ) );
+  newFields.append( QgsField( QStringLiteral( "extremum" ), QMetaType::Type::QString ) );
```

Comment:
``extremum`` makes me think that it would be the value. Maybe ?
```suggestion
  newFields.append( QgsField( QStringLiteral( "extremum_type" ), QMetaType::Type::QString ) );
```
Or maybe create 2 fields "minimum" and "maximum", and only fill the appropriate one ?

Reply:
Ok, I've gone with extremum_type, and also updated the recent raster min/max point algorithm too